app/subscriber/src/store/hooks/subscriber/useNavigateAndScroll.ts

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+export const useNavigateAndScroll = () => {
+  const navigate = useNavigate();
+
+  const controller = React.useMemo(
```

Comment:
Since this hook only returns a single function you don't need to return an object.  Do the following.

```
return React.useCallback((path: string) => {
navigate(path);
ref.current?.scrollTo(0, 0);
}, [navigate]);
```